bubo/bot_commands.py

```diff
@@ -627,6 +642,40 @@ async def _list_rooms(self, spaces: bool = False):
         text += "".join(rooms_list)
         return text
 
+    async def _pindora(self):
+        if not self.config.pindora_enabled:
+            await send_text_to_room(self.client, self.room.room_id, help_strings.HELP_PINDORA_DISABLED)
+            return
+
+        if not await self._ensure_pindora_user():
+            return
+
+        if self.args:
+            subcommand = self.args[0]
+            if subcommand == "create":
+                name = None
+                try:
+                    name = self.args[1]
+                except:
+                    pass
+
+                if not name:
+                    await send_text_to_room(self.client, self.room.room_id, help_strings.HELP_KEYS)
+                    return
+
+                hours = None
+                try:
+                    hours = self.args[2]
+                except:
+                    pass
+
+                key, magic_url = create_new_key(self.config.pindora_id, self.config.pindora_token, name, hours=hours, pindora_timezone=self.config.pindora_timezone)
```

Comment:
Since will potentially raise later in `response.raise_for_status()` it needs to be surrounded by a try/catch, otherwise a network failure will crash bubo.